conda/gateways/disk/create.py

```diff
@@ -246,3 +254,53 @@ def compile_pyc(python_exe_full_path, py_full_path):
                   "  stderr: %s\n"
                   "  rc: %d", command, stdout, stderr, rc)
         raise RuntimeError()
+
+
+def get_json_content(path_to_json):
+    if isfile(path_to_json):
+        try:
+            with open(path_to_json, "r") as f:
+                json_content = json.load(f)
+        except json.decoder.JSONDecodeError:
+            json_content = {}
+    else:
+        json_content = {}
+    return json_content
+
+
+def create_private_envs_meta(pkg, prefix):
+    # type: (str, str -> ()
+    path_to_conda_meta = join(context.root_prefix, "conda-meta")
+    path_to_private_envs = join(path_to_conda_meta, "private_envs")
+
+    if not isdir(path_to_conda_meta):
+        os.mkdir(path_to_conda_meta)
+
+    private_envs_json = get_json_content(path_to_private_envs)
+    private_envs_json[pkg] = prefix
+    with open(path_to_private_envs, "w") as f:
+        json.dump(private_envs_json, f)
+
+
+def remove_private_envs_meta(pkg):
+    path_to_conda_meta = join(context.root_prefix, "conda-meta")
+    path_to_private_envs = join(path_to_conda_meta, "private_envs")
+
+    private_envs_json = get_json_content(path_to_private_envs)
+    if pkg in private_envs_json.keys():
+        private_envs_json.pop(pkg)
+    if private_envs_json == {}:
+        rm_rf(path_to_private_envs)
+    else:
+        with open(path_to_private_envs, "w") as f:
+            json.dump(private_envs_json, f)
+
+
+def create_private_pkg_entry_point(target_path, python_full_path, private_env_prefix,
+                                   app_name):
+    entry_point = private_pkg_entry_point_template % {"private_env_path": private_env_prefix,
+                                                      "app_name": app_name}
+    with open(target_path, "w") as fo:
+        fo.write('#!%s\n' % python_full_path)
+        fo.write(entry_point)
+    chmod(target_path, 0o755)
```

Comment:
I know we have this `chmod( , 0o755)` somewhere else.  I don't like it because 755 might be wrong.  What we really want here is the equivalent of bash `chmod +x`.  Look at the `make_writable()` function in `conda.gateways.disk.permissions` for an example.  Or maybe just add a `make_executable()` function there.